flink/src/test/java/org/apache/iceberg/flink/TestIcebergConnector.java

```diff
@@ -0,0 +1,136 @@
+        .newInstance()
+        .useBlinkPlanner()
+        .inBatchMode()
+        .build();
+
+    tEnv = TableEnvironment.create(settings);
+    tEnv.getConfig().getConfiguration()
+        .set(CoreOptions.DEFAULT_PARALLELISM, 1)
+        .set(FlinkTableOptions.TABLE_EXEC_ICEBERG_INFER_SOURCE_PARALLELISM, false);
+  }
+
+  @Test
+  public void testHadoopCatalog() {
+    Map<String, String> tableProps = Maps.newHashMap();
+    tableProps.put("connector", "iceberg");
+    tableProps.put("catalog-name", "test-hadoop");
+    tableProps.put("catalog-type", "hadoop");
+    tableProps.put("catalog-database", "local_db");
+    tableProps.put(CatalogProperties.WAREHOUSE_LOCATION, warehouseRoot());
+
+    sql("CREATE TABLE hadoop_table (id BIGINT, data STRING) WITH %s", toWithClause(tableProps));
+    sql("INSERT INTO hadoop_table VALUES (1, 'aaa'), (2, 'bbb'), (3, 'ccc')");
+    Assert.assertEquals("Should have expected rows",
+        Lists.newArrayList(Row.of(1L, "aaa"), Row.of(2L, "bbb"), Row.of(3L, "ccc")),
+        sql("SELECT * FROM hadoop_table"));
+
+    // Drop and create it again.
+    sql("DROP TABLE hadoop_table");
+    sql("CREATE TABLE hadoop_table (id BIGINT, data STRING) WITH %s", toWithClause(tableProps));
+    Assert.assertEquals("Should have expected rows",
+        Lists.newArrayList(Row.of(1L, "aaa"), Row.of(2L, "bbb"), Row.of(3L, "ccc")),
+        sql("SELECT * FROM hadoop_table"));
+
+    sql("DROP TABLE hadoop_table");
+  }
+
+  @Test
+  public void testHiveCatalog() {
+    Map<String, String> tableProps = Maps.newHashMap();
+    tableProps.put("connector", "iceberg");
+    tableProps.put("catalog-name", "test-hive");
+    tableProps.put("catalog-type", "hive");
+    tableProps.put("catalog-database", "default");
+    tableProps.put(CatalogProperties.URI, FlinkCatalogTestBase.getURI(hiveConf));
+    tableProps.put(CatalogProperties.WAREHOUSE_LOCATION, warehouseRoot());
+
+    sql("CREATE TABLE hive_table(id BIGINT, data STRING) WITH %s", FlinkCatalogTestBase.toWithClause(tableProps));
+    sql("INSERT INTO hive_table VALUES (1, 'AAA'), (2, 'BBB'), (3, 'CCC')");
+    Assert.assertEquals("Should have the expected rows",
+        Lists.newArrayList(Row.of(1L, "AAA"), Row.of(2L, "BBB"), Row.of(3L, "CCC")),
+        sql("SELECT * FROM hive_table"));
+
+    // Drop and create it again.
+    sql("DROP TABLE hive_table");
+    sql("CREATE TABLE hive_table (id BIGINT, data STRING) WITH %s", toWithClause(tableProps));
+    Assert.assertEquals("Should have expected rows",
+        Lists.newArrayList(Row.of(1L, "AAA"), Row.of(2L, "BBB"), Row.of(3L, "CCC")),
+        sql("SELECT * FROM hive_table"));
+
+    sql("DROP TABLE hive_table");
+  }
+
+  private String toWithClause(Map<String, String> props) {
+    return FlinkCatalogTestBase.toWithClause(props);
+  }
+
+  private String warehouseRoot() {
+    return String.format("file://%s", warehouse.getRoot().getAbsolutePath());
+  }
+
+  protected List<Row> sql(String query, Object... args) {
```

Comment:
`FlinkTestBase` already has a `sql` method. it seems that the only diff is the extra call of `tableResult.await()` here. Is it necessary considering we are calling `tableResult.collect()`?

Reply:
+1 on double checking whether it's needed at all. When I've removed the test still passed.
If there is an objective reason why this needed the `@Override` must be added.

Reply:
Sounds reasonable.